src/FilePathsBase.jl

```diff
@@ -81,6 +95,7 @@ root(path::AbstractPath) = error("`root` not implemented.")
 drive(path::AbstractPath) = error("`drive` not implemented.")
 
 Base.convert(::Type{AbstractPath}, x::AbstractString) = Path(x)
+ispathtype(::Type{T}, x::AbstractString) where {T<:AbstractPath} = false
```

Comment:
Would it make sense to allow regular expressions in addition to strings?

Reply:
Currently, Path constructors don't take a regex or glob pattern, so probably not. To clarify, `x` is just a string representation of a path similar to the `convert` method. Individual path types may want to implement this method using regex matching internally though.

Reply:
Okay, I think I misunderstood then. I thought this was for identifying a path of a particular type based on the given pattern.